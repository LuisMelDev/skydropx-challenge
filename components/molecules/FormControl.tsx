import React, { ChangeEventHandler, HTMLInputTypeAttribute } from "react";
import { FormHelperText, Input, Label } from "..";

interface Props {
  labelText: string;
  idInput: string;
  value: any;
  onChange: ChangeEventHandler;
  typeInput: HTMLInputTypeAttribute;
  placeholder?: string;
  textError: string;
  nameInput?: string;
  size?: "full" | "50";
}

export const FormControl = ({
  labelText,
  idInput,
  value,
  onChange,
  typeInput,
  placeholder = "",
  textError = "",
  nameInput = "",
  size = "full",
}: Props) => {
  return (
    <div className={`${size === "full" ? "full block" : "w-5/12 inline-block"} mb-3`}>
      <Label
        htmlFor={idInput}
        text={labelText}
        className="text-sm font-light block"
      />
      <Input
        id={idInput}
        value={value}
        onChange={onChange}
        type={typeInput}
        placeholder={placeholder}
        name={nameInput}
        className="bg-slate-200 outline-none focus:bg-slate-50 focus:shadow font-normal h-10 rounded-lg py-1 px-2 text-md block w-full"
      />
      <FormHelperText text={textError} />
    </div>
  );
};
