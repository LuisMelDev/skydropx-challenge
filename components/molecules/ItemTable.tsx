import React, { MouseEventHandler } from "react";
import { Tab, Td } from "..";

interface ItemTableProps{
    rate: any;
    recomended: boolean;
    onClick: MouseEventHandler<HTMLTableRowElement>
}

export const ItemTable = ({
    rate,
    recomended = false,
    onClick
}: ItemTableProps) => {
    const { provider, service_level_name, days, total_pricing } = rate;
    return (
        <tr className='hover:bg-gray-100 active:bg-gray-200 cursor-pointer transition ease-in-out delay-80' onClick={onClick} >
            <Td>
                <div className='flex items-center'>
                    <p>{provider}</p>
                </div>
            </Td>
            <Td>
                <p>{service_level_name}</p>
            </Td>
            <Td>
                <p>{days} dias</p>
            </Td>
            <Td>
                <p>{total_pricing} MXN</p>
            </Td>
            <Td>{recomended ? <Tab text='Mejor Opcion' /> : null}</Td>
        </tr>
    );
};
