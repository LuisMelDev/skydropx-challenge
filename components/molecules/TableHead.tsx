import React from "react";
import { Th } from "..";

export const TableHead = ({ heads }: { heads: string[] }) => {
    return (
        <thead>
            <tr>
                {heads.map((name) => (
                    <Th name={name} key={name} />
                ))}
            </tr>
        </thead>
    );
};

