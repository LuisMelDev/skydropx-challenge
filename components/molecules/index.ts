export * from "./Nav"
export * from "./FormControl"