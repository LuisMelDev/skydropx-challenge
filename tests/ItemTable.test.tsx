import React from "react";
import { shallow } from "enzyme";
import { ItemTable } from "../components/molecules/ItemTable";

const rate = {
    created_at: "2022-03-19T23:03:42.909-06:00",
    updated_at: "2022-03-19T23:03:42.945-06:00",
    amount_local: "134.0",
    currency_local: "MXN",
    provider: "CARSSA",
    service_level_name: "Nacional",
    service_level_code: "NACIONAL",
    service_level_terms: null,
    days: 10,
    duration_terms: null,
    zone: null,
    arrives_by: null,
    out_of_area: false,
    out_of_area_pricing: "0.0",
    total_pricing: "134.0",
    is_ocurre: false,
};

describe("Pruebas <ItemTable>", () => {
    const onClick = jest.fn();
    let wrapper = shallow(<ItemTable rate={rate} recomended onClick={onClick} />);

    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

});
