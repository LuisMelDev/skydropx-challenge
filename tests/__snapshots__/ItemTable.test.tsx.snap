// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <ItemTable> debe de mostrarse correctamente 1`] = `
<tr
  className="hover:bg-gray-100 active:bg-gray-200 cursor-pointer transition ease-in-out delay-80"
  onClick={[MockFunction]}
>
  <Td>
    <div
      className="flex items-center"
    >
      <p>
        CARSSA
      </p>
    </div>
  </Td>
  <Td>
    <p>
      Nacional
    </p>
  </Td>
  <Td>
    <p>
      10
       dias
    </p>
  </Td>
  <Td>
    <p>
      134.0
       MXN
    </p>
  </Td>
  <Td>
    <Tab
      text="Mejor Opcion"
    />
  </Td>
</tr>
`;
